### PR TITLE
feat: add room page

### LINKED ISSUE #153: Implement base room page design

### PR DESCRIPTION
**Pull request by junlong4321**
Resolve #153

### CODE REVIEW

File: frontend/src/pages/RoomPage.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+export const RoomPage = () => {
+  return (
+    <Grid container>
+      <Grid item xs={12} sx={{ height: "10px", bgcolor: "primary.500" }}></Grid>
+      <Grid item xs={12}>
+        <Grid
+          container
+          item
+          xs={12}
+          sx={{ p: "1%", justifyContent: "space-between" }}
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:25]:
```suggestion
          sx={{ py: 2, px: 3, justifyContent: "space-between" }}
```

File: frontend/src/pages/RoomPage.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        >
+          <Stack direction="row" spacing={1}>
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:27]:
```suggestion
<Stack
            direction="row"
            spacing={1}
            sx={{ display: "flex", alignItems: "center" }}
          >
```

File: frontend/src/pages/RoomPage.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                width: "36px",
+                height: "36px",
+                bgcolor: "primary.A700",
+              }}
+            />
+          </Stack>
+          <StyledButton label={"Leave Room"} sx={{ bgcolor: "red.500" }} />
+        </Grid>
+        <Divider />
+        <Stack direction="row" spacing={3} sx={{ p: "1%" }}>
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:46]:
```suggestion
        <Stack direction="row" spacing={3} sx={{ p: 3 }}>
```

File: frontend/src/pages/RoomPage.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+          <Grid item xs={6}>
+            <Paper elevation={1} sx={{ p: "4%" }}>
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:48]:
```suggestion
          <Grid item xs={6}>
            <Paper elevation={1} sx={{ p: 3 }}>
```

File: frontend/src/pages/RoomPage.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+              <Stack direction="column" spacing={3}>
+                <Grid item xs={12}>
+                  <Typography variant="h5" sx={{ textAlign: "center" }}>
+                    Problem Title
+                  </Typography>
+                </Grid>
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:54]:
```suggestion
                <Typography variant="h5" sx={{ textAlign: "center" }}>
                  Problem Title
                </Typography>
```

File: frontend/src/pages/RoomPage.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+              }}
+            />
+          </Stack>
+          <StyledButton label={"Leave Room"} sx={{ bgcolor: "red.500" }} />
+        </Grid>
+        <Divider />
+        <Stack direction="row" spacing={3} sx={{ p: "1%" }}>
+          <Grid item xs={6}>
+            <Paper elevation={1} sx={{ p: "4%" }}>
+              <Stack direction="column" spacing={3}>
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:49]:
```suggestion
              <Stack direction="column" spacing={2}>
```

File: frontend/src/pages/RoomPage.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+                </Typography>
+                <Typography variant="body2" sx={{ textAlign: "justify" }}>
+                  Aliquam sodales, odio quis sagittis dignissim, nibh ipsum
+                  tempus nibh, ut ullamcorper nisl quam et elit. Phasellus
+                  sagittis libero nisl, vitae convallis nibh tincidunt quis.
+                </Typography>
+                <Accordion
+                  sx={{
+                    boxShadow: "none",
+                    "& .MuiAccordionSummary-root": {
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:81]: Might want to recheck the encoding of the `&` character.

File: frontend/src/pages/RoomPage.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+              </Stack>
+            </Paper>
+          </Grid>
+          <Grid item xs={6}>
+            <Paper elevation={1} sx={{ p: "3%" }}>
+              <Grid item xs={12}>
+                <Typography variant="h5" sx={{ textAlign: "center" }}>
+                  Editor
+                </Typography>
+              </Grid>
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:106]:
```suggestion
              <Typography variant="h5" sx={{ textAlign: "center" }}>
                Editor
              </Typography>
```

File: frontend/src/pages/RoomPage.tsx [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+                      Suspendisse malesuada lacus ex, sit amet blandit leo
+                      lobortis eget.
+                    </Typography>
+                  </AccordionDetails>
+                </Accordion>
+              </Stack>
+            </Paper>
+          </Grid>
+          <Grid item xs={6}>
+            <Paper elevation={1} sx={{ p: "3%" }}>
```
Comment (dextertanyj) [inline frontend/src/pages/RoomPage.tsx:101]:
```suggestion
            <Paper elevation={1} sx={{ p: 3 }}>
```

**Review by dextertanyj — APPROVED** [review]
lgtm

File: frontend/src/components/Question.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+  const handleChange = () => {
+    setExpanded(!expanded);
+  };
+
+  return expanded ? (
+    <Grid item xs={4}>
+      <Paper elevation={1} sx={{ p: 3 }}>
+        <Stack direction="column" spacing={3}>
+          <Center>
```
Comment (junlong4321) [inline frontend/src/components/Question.tsx:29]: Is there a way to position this such that the typography is centered, but the icon button is flex-end?

**Pull request merged**